core-api/src/main/java/org/silverpeas/core/util/DateUtil.java

```diff
@@ -709,23 +717,64 @@ public static String formatTime(Calendar calend) {
+   * into account if null.
    * @return the formatted String.
    */
-  public static String formatDateAndTime(ZonedDateTime dateTime, String language) {
-    if (dateTime == null) {
+  public static String getOutputDateAndHour(java.time.temporal.Temporal temporal, String language,
+      ZoneId zoneIdReference) {
+    if (temporal == null) {
       return null;
     }
-    boolean sameOffsetAsPlatform = dateTime.getOffset().getTotalSeconds() ==
-        OffsetDateTime.now().getOffset().getTotalSeconds();
-    final String datePattern = getDateOutputFormat(language).getPattern();
-    final String timePattern = getHourOutputFormat(language).getPattern();
-    final String formattedDateTime =
-        dateTime.format(DateTimeFormatter.ofPattern(datePattern + " " + timePattern));
-    return sameOffsetAsPlatform
-        ? formattedDateTime
-        : formattedDateTime + " (" + dateTime.getZone().getId() + ")";
+    return (getOutputDate(temporal, language) + " " +
+        getOutputHour(temporal, language, zoneIdReference)).trim();
+  }
+
+  /**
+   * Formats the time of an {@link OffsetDateTime} instance with the right given language.
+   * @param temporal an {@link OffsetDateTime} instance.
+   * @param language the language for formatting.
+   * @param zoneIdReference the identifier of the zone which represents the reference, not taken
+   * into account if null.
+   * @return the formatted String.
+   */
+  public static String getOutputHour(java.time.temporal.Temporal temporal, String language,
+      ZoneId zoneIdReference) {
+    final DateTimeFormatter hourFormat = ofPattern(getHourOutputFormat(language).getPattern());
+    final Mutable<ZoneId> zoneIdIfDifferentToReference = Mutable.empty();
+    final String hour = new FilterByType(temporal)
+        .matchFirst(LocalDateTime.class::equals, t -> hourFormat.format((LocalDateTime) t))
+        .matchFirst(OffsetDateTime.class::equals, t -> {
+          final OffsetDateTime o = (OffsetDateTime) t;
+          if (zoneIdReference != null && o.getOffset().getTotalSeconds() !=
+              OffsetDateTime.now(zoneIdReference).getOffset().getTotalSeconds()) {
+            zoneIdIfDifferentToReference.set(o.toZonedDateTime().getZone());
+          }
+          return hourFormat.format(o);
+        })
+        .matchFirst(ZonedDateTime.class::equals, t -> {
+          final ZonedDateTime o = (ZonedDateTime) t;
+          if (zoneIdReference != null && o.getOffset().getTotalSeconds() !=
+              ZonedDateTime.now(zoneIdReference).getOffset().getTotalSeconds()) {
+            zoneIdIfDifferentToReference.set(o.getZone());
+          }
+          return hourFormat.format(o);
+        })
+        .result()
+        .orElse("");
+    return !zoneIdIfDifferentToReference.isPresent()
+        ? hour
+        : hour + " (" + zoneIdIfDifferentToReference.get().getId() + ")";
+  }
+
+  public static String getOutputZoneId(java.time.temporal.Temporal temporal, String language) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "language". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1172)